lib/ocpp/v201/charge_point.cpp

```diff
@@ -3760,6 +3795,10 @@ std::map<SetVariableData, SetVariableResult>
 ChargePoint::set_variables(const std::vector<SetVariableData>& set_variable_data_vector) {
     // set variables and allow setting of ReadOnly variables
     const auto response = this->set_variables_internal(set_variable_data_vector, true);
+
+    // process all triggered monitors
+    this->monitoring_updater.process_triggered_monitors();
```

Comment:
Shouldnt this be executed inside  `set_variables_internal` or `handle_variables_changed` to also cover internal changes of variable values?